src/main/java/com/mraof/minestuck/item/weapon/MusicPlayerItem.java

```diff
@@ -0,0 +1,234 @@
+				{
+					MusicPlayerPacket packet;
+					if(musicPlayingCap.getCassetteType() == EnumCassetteType.NONE)
+					{
+						packet = MusicPlayerPacket.createPacket(playerIn, cassette.cassetteID, volume, pitch);
+						musicPlayingCap.setMusicPlaying(musicPlayer, cassette.cassetteID);
+					} else
+					{
+						packet = MusicPlayerPacket.createPacket(playerIn, EnumCassetteType.NONE, 0, 0);
+						musicPlayingCap.setMusicPlaying(ItemStack.EMPTY, EnumCassetteType.NONE);
+					}
+					MSPacketHandler.sendToTrackingAndSelf(packet, playerIn);
+				}
+			}
+			//open the GUI if right-clicked
+			else
+			{
+				MusicPlayerPacket packet = MusicPlayerPacket.createPacket(playerIn, EnumCassetteType.NONE, 0, 0);
+				musicPlayingCap.setMusicPlaying(ItemStack.EMPTY, EnumCassetteType.NONE);
+				MSPacketHandler.sendToTrackingAndSelf(packet, playerIn); //This will stop the music before opening the GUI
+				
+				NetworkHooks.openGui((ServerPlayer) playerIn, new SimpleMenuProvider((pContainerId, pInventory, pPlayer) ->
+						new CassetteContainerMenu(pContainerId, pInventory, itemStackHandlerMusicPlayer, musicPlayer),
+						new TranslatableComponent(TITLE)));
+			}
+			return InteractionResultHolder.success(musicPlayer);
+		}
+		return super.use(level, playerIn, handIn);
+	}
+	
+	@Override
+	public void inventoryTick(ItemStack stack, Level level, Entity entityIn, int itemSlot, boolean isSelected)
+	{
+		super.inventoryTick(stack, level, entityIn, itemSlot, isSelected);
+		ItemStack itemInMusicPlayer = getItemStackHandlerMusicPlayer(stack).getStackInSlot(0);
+		setHasCassette(stack, !itemInMusicPlayer.isEmpty());
+	}
+	
+	@SubscribeEvent
+	public static void playerTick(TickEvent.PlayerTickEvent tickEvent)
+	{
+		if(tickEvent.side.isServer() && tickEvent.phase == TickEvent.Phase.END)
+		{
+			Player player = tickEvent.player;
+			IMusicPlaying musicPlayingCap = getMusicPlaying(player);
+			
+			if(!(player.getItemInHand(InteractionHand.MAIN_HAND) == musicPlayingCap.getCurrentMusicPlayer() ||
+					player.getItemInHand(InteractionHand.OFF_HAND) == musicPlayingCap.getCurrentMusicPlayer()) ||
+					musicPlayingCap.getCurrentMusicPlayer().isEmpty())
+			{
+				if(musicPlayingCap.getCassetteType() != EnumCassetteType.NONE)
+				{ //If the Cassette player isn't in hand and is playing a music, stop it
+					MusicPlayerPacket packet = MusicPlayerPacket.createPacket(player, EnumCassetteType.NONE, 0, 0);
+					musicPlayingCap.setMusicPlaying(ItemStack.EMPTY, EnumCassetteType.NONE);
+					MSPacketHandler.sendToTrackingAndSelf(packet, player);
+				}
+			}
+			
+			if(player.level.getGameTime() % 50 == 0 && musicPlayingCap.getCassetteType() != EnumCassetteType.NONE)
+			{
+				EnumCassetteType.EffectContainer effectContainer = musicPlayingCap.getCassetteType().getEffectContainer();
+				if(!effectContainer.onHit()) //Apply the cassette buff every 50 ticks, if there is any
+					player.addEffect(effectContainer.effect().get());
+			}
```

Comment:
I dont think its worth changing, but I am going to note that the weapon continues to give the user buffs(if the cassette does that) even after the music has stopped playing

Reply:
I would have liked to loop the music, but the soundinstance class im using do not allow it.

Reply:
Spontaneous thought: should a cassette player be able to loop in the first place? At some point further down the line, we could try implementing cassettes so that they need to be rewound to play from the beginning.